Source/MQTTnet/Server/MqttClientSessionsManager.cs

```diff
@@ -224,12 +224,28 @@ async Task TryProcessNextQueuedApplicationMessageAsync(CancellationToken cancell
                     await _retainedMessagesManager.HandleMessageAsync(sender?.ClientId, applicationMessage).ConfigureAwait(false);
                 }
 
+                var deliveryCount = 0;
+
                 foreach (var clientSession in _sessions.Values)
                 {
                     clientSession.EnqueueApplicationMessage(
                         applicationMessage,
                         sender?.ClientId,
                         false);
+
+                    deliveryCount++;
```

Comment:
this makes no sense we at least have to check if this session is subscribed to that message

Reply:
How can I do that? 😅

Reply:
@JanEggers By the way, that's what I meant with the other comment already:

> However, EnqueueApplicationMessage doesn't return anything, it's a void type method. I will implement this and let you check again.

Reply:
ahh :) 

you could add a task completion source to each message in ApplicationMessagesQueue
return the task and complete it once the session checked it.

then wait for all sessions in the server and count the transmissions.

I would cancel this PR as its a waste of performance and makes no sense in general.

If you want to know a message was delivered from A to B. B has to send a new message aka response to A

Reply:
@chkr1011 What do you think? Should we close this?

> I would cancel this PR as its a waste of performance and makes no sense in general.

> If you want to know a message was delivered from A to B. B has to send a new message aka response to A

That's true. Honestly makes no sense after thinking about this a bit more :) I agree with you, we should close this. Let's wait for @chkr1011 to have his opinion here as well.

Note: I just implemented @cshark-inator's request to get a bit more things done here :)

Reply:
In my opinion this should be fairly easy. The method _EnqueueApplicationMessage_ can easily return a boolean because it uses it internally as well (checkSubscriptionsResult.IsSubscribed). We can use this to increase the delivery count. If this results in 0 we can invoke the handler.

In my opinion this is not a killer feature but if someone needs that approach we should add it. There is be no noticeable performance impact because we are only increasing an integer number.

@JanEggers I assume you are referring to a delivery count which includes the transmit to the client. I only refer to that fact that a message was subscribed and will be processed in the near future. A use case could be identification of "dead" messages which are not subscribed by anyone. So a valid feature request in my opinion.

Reply:
Ok, sounds good. I will check this and make some changes there.